src/Codeception/Test/Loader/Cest.php

```diff
@@ -50,6 +52,27 @@ function ($v) {
                         },
                         $rawExamples
                     );
+                }
+
+                // dataprovider Annotation
+                $dataMethod = Annotation::forMethod($testClass, $method)->fetch('dataprovider');
+                if(!empty($dataMethod)) {
+                    if (false === is_callable([$unit, $dataMethod])) {
+                        throw new TestParseException(
+                            $file, "DataProvider $dataMethod for $testClass->$method is invalid or not callable."
+                            . PHP_EOL .
+                            "Make sure that the dataprovider method is a public static function."
+                        );
+                    }
+                    // allow to mix axample and dataprovider annotations
+                    if(count($examples)) {
+                        $examples = array_merge($examples, $testClass::$dataMethod());
+                    } else {
+                        $examples = $testClass::$dataMethod();
```

Comment:
I agree on this. I was running into this "problem" when using the dataprovider extension. Other than that, i make full use of this extension at my current project so i would love to see this extension being imported into the core asap